src/main/java/software/aws/neptune/NeptuneDriver.java

```diff
@@ -76,62 +68,75 @@ public boolean acceptsURL(final String url) throws SQLException {
         if (!acceptsURL(url)) {
             return null;
         }
-        final java.sql.Connection connection;
         try {
-            final String language = getLanguage(url, CONN_STRING_PATTERN);
-            final Properties properties =
-                    parsePropertyString(getPropertyString(url, CONN_STRING_PATTERN), firstPropertyKey(language));
-            if (info != null) {
-                properties.putAll(info);
-            }
-            final ConnectionProperties connectionProperties = connectionProperties(language, properties);
-            connection = (java.sql.Connection) connectionMap.get(language)
-                    .getConstructor(ConnectionProperties.class)
-                    .newInstance(connectionProperties);
-            int connectionTimeoutSeconds = connectionProperties.getConnectionTimeoutMillis() / 1000;
-            if (connectionTimeoutSeconds <= 0) {
-                connectionTimeoutSeconds = 1;
-            }
-
-            if (connection.isValid(connectionTimeoutSeconds)) {
-                return connection;
-            }
+            final DriverConnectionFactory connectionFactory = new DriverConnectionFactory();
+            return connectionFactory.getConnection(url, info);
         } catch (final Exception e) {
             if (e instanceof SQLException) {
                 throw (SQLException) e;
             }
             LOGGER.error("Unexpected error while creating connection:", e);
             return null;
         }
-        throw new SQLException("Error, failed to connect to database. Check endpoint, port, and configuration.");
     }
 
-    private ConnectionProperties connectionProperties(final String language, final Properties properties)
-            throws SQLException {
-        // TODO AN-550: Switch to map with class that holds Conn properties, key, query executor, etc.
-        if (OPENCYPHER.equalsIgnoreCase(language)) {
-            return new OpenCypherConnectionProperties(properties);
-        }
-        if (GREMLIN.equalsIgnoreCase(language) || SQL_GREMLIN.equalsIgnoreCase(language)) {
-            return new GremlinConnectionProperties(properties);
-        }
-        if (SPARQL.equalsIgnoreCase(language)) {
-            return new SparqlConnectionProperties(properties);
-        }
-        return new OpenCypherConnectionProperties(properties);
-    }
+    private class DriverConnectionFactory {
+        private final Pattern connStringPattern = Pattern.compile(CONN_STRING_PREFIX + "(\\w+)://(.*)");
```

Comment:
Not really a problem nor is it necessarily within the scope of this ticket, but IMO regex should be a last resort.

It seems like we can just use a `String.contains` or something along those lines to determine the connection?

Either way it doesn't need to be addressed here.

Reply:
This was taken from existing implementation, and I thought it was used as a last resort? @lyndonb-bq did you try others?